### PR TITLE
SP-63 Feature/improve admin actions extensibility

### PR DESCRIPTION
**Pull request by tomdonarski**
### **What?**
[SP-63](https://upsidelab.atlassian.net/browse/SP-63)
Provide builder classes to configure actions in a couple of resources (e.g order, product, user), administered in the admin panel.

### **Why?**
In order for the extensions to be able to configure/extend via code, instead of using deface (external gem).

### **How?**
Based on the approach implemented in #274 

[SP-63]: https://upsidelab.atlassian.net/browse/SP-63

### CODE REVIEW

File: app/models/spree/admin/actions/product_default_actions_builder.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+            build
+
+          root.add(action)
+        end
+
+        def external_preview_config
+          {
+            icon_name: 'view.svg',
+            name: 'admin.utilities.preview',
+            url: ->(resource) { spree_storefront_resource_url(resource) },
```
[the remark this answers is not in the document]
Comment (rafalcymerys) [reply app/models/spree/admin/actions/product_default_actions_builder.rb:27]: Yeah, this link depends on the spree_frontend gem being installed, which is not necessarily true (this route doesn't exist when a next.js/vue storefront is used, or when someone uses Spree only via APIs)

[the remark this answers is not in the document]
Comment (tomdonarski) [reply app/models/spree/admin/actions/product_default_actions_builder.rb:27]: Does this mean that for shops running e.g. API only this feature is broken? 🤔

File: app/views/spree/admin/variants/index.html.erb [hunk 2]
```diff
@@ -64,8 +64,18 @@
...
+        action.text,
+        action.url(@product),
+        class: action.classes,
+        icon: action.icon_name,
+        id: action.id,
+        target: action.target,
+        data: action.data
+      ) %>
+    <% end %>
     <%= button_link_to (@deleted == '1' ? Spree.t(:show_active) : Spree.t(:show_deleted)), spree.admin_product_variants_url(@product, { q: { deleted_at_null: @deleted == '1' ? "0" : "1" } }), { class: 'btn-light', icon: 'filter.svg' } %>
```
Comment (tomdonarski) [inline app/views/spree/admin/variants/index.html.erb:79]: This is the last one standing. I haven't figured out how to tackle it yet.

File: app/models/spree/admin/actions/product_properties_default_actions_builder.rb [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+            build
+
+          root.add(action)
+        end
+
+        def add_product_properties_config
+          {
+            icon_name: 'add.svg',
+            key: :add_product_properties,
+            url: 'javascript:;',
```
Comment (tomdonarski) [inline app/models/spree/admin/actions/product_properties_default_actions_builder.rb:47]: ⚠️ This isn't working on `main`. I'll take a look what the proper path would be.

**Pull request merged**